pyomo/contrib/iis/mis.py

```diff
@@ -287,7 +287,7 @@ def _constraint_generator():
     except:
         results = None
 
-    if pyo.check_optimal_termination(results):
+    if (results is not None) and pyo.check_optimal_termination(results):
```

Comment:
Question: would it better to not swallow all exceptions from the `solve()` above and instead use `load_solutions=False` in the call?  Then results should never be `None`?

Reply:
Probably. Would that still not raise an exception if the solver returns a non-zero exit code? I think that's why I've implemented these as blanket `try / except` blocks.

Reply:
OK.  I would just be worried about the bare `except:` -- that could be suppressing any number of errors and there would be no way to know what was going on.